### PR TITLE
Team Round Robin

### PR DESCRIPTION
**Pull request by nalex1211**
I added the ability to create team round robin tournament. For now you can only create it, you can't register a team there, and you can edit it. Also the database had to be changed a little bit for that tournament. Also fixed the edit problem. Now if field is required, it will show it and won't let you save changes(before that it was possible to press "save changes"  button but nothing would happen exept being redirected to the ghome page)

### CODE REVIEW

**Review by rhourani — APPROVED** [review]

File: frontend/locales/fi.json [hunk 1]
```diff
@@ -210,9 +212,12 @@
     "description": "Kuvaus",
     "select_tournament_type": "Valitse turnaustyypi",
     "round_robin": "Kaikki vastaan kaikki",
+    "team_round_robin": "Joukkue Round Robin",
```
Comment (metsoila) [inline frontend/locales/fi.json:215]: This translation could be checked. Round robin is translated above in finnish "free for all" or "everyone against everyone". Maybe just ask Markus in Sprint review and fix later unless this is the correct translation

File: frontend/locales/fi.json [hunk 2]
```diff
@@ -241,6 +246,10 @@
   "edit_tournament_labels": {
     "edit_tournament": "Muokkaa turnausta"
   },
+  "team_info_labels": {
+    "team_name": "Joukkueen nimet",
```
Comment (metsoila) [inline frontend/locales/fi.json:250]: team's name: joukkueen nimi
teams names (multiple): joukkueiden nimet

I dont know context so I dont know which this should be but I would guess the above one(?)

File: frontend/locales/fi.json [hunk 3]
```diff
@@ -272,6 +281,7 @@
   },
   "types": {
     "round_robin": "Kaikki vastaan kaikki",
+    "team_round_robin": "Joukkue Round Robin",
```
Comment (metsoila) [inline frontend/locales/fi.json:284]: same thing as the other comment. Could be fixed later and merge this if no need for functionality changes

**Review by metsoila — APPROVED** [review]
Can merge this, created a ticket for the translations. I think they can be fixed all in one ticket after functionality is done

**Pull request merged**